buildlib/tools/test_mad.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+set -exE -o pipefail
+
+IMAGE="rdmz-harbor.rdmz.labs.mlnx/ucx/x86_64/rhel8.2/builder:mofed-5.0-1.0.0.0"
+cd "$BUILD_SOURCESDIRECTORY"
+
+node_setup() {
+    funcname
+    sudo chmod 777 /dev/infiniband/umad*
+}
+
+build_ucx() {
+    funcname
+    ./autogen.sh
+    ./contrib/configure-release \
+        --prefix="$PWD"/install \
+        --with-mad \
+        --without-valgrind \
+        --without-go \
+        --without-java
+    make -s -j"$(nproc)"
+    make install
+}
+
+build_ucx_in_docker() {
+    docker run --rm \
+        --name ucx_build_"$BUILD_BUILDID" \
+        -v "$PWD":"$PWD" -w "$PWD" \
+        -v /hpc/local:/hpc/local \
+        $IMAGE \
+        bash -c "source ./buildlib/tools/test_mad.sh && build_ucx"
+
+    sudo chown -R swx-azure-svc:ecryptfs "$PWD"
+}
```

Comment:
can we do it by Azure "container:" field, like in other build jobs, instead of running docker manually?

Reply:
IMO, we should separate business logic from flow, having all the logic in this script helps by allowing easy way of running the tests without having to reverse engineer the AZP yaml.

Reply:
Had to resort to the manual Docker run commands to run the service container in a detached mode. 
To build UCX in Docker we can indeed use a separate job with a "container:" field. 
As Daniel mentioned, this would affect the code's readability.

Reply:
usually our build jobs are in a separate task with "container:" tag. so i think for consistency this one should do it as well.